### PR TITLE
Specify default token list

### PR DESCRIPTION
**Pull request by fleupold**
Because it's more convenient

### Test Plan

yarn hardhat trade --network rinkeby

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -13,7 +13,18 @@ export class ChainUtils {
...
+      throw `Unexpected network ${network.config.chainId}`;
+    }
+  }
+
+  static defaultTokenList(chain: Chain): string {
+    switch (chain) {
+      case Chain.MAINNET:
+        return "https://raw.githubusercontent.com/Uniswap/token-lists/master/test/schema/bigexample.tokenlist.json";
+      case Chain.RINKEBY:
+        return "https://raw.githubusercontent.com/compound-finance/token-list/master/compound.tokenlist.json";
```
Comment (bh2smith) [inline src/utils.ts:25]: Nice! I was looking for a good rinkeby token list!

**Review by bh2smith — APPROVED** [review]

File: src/make_trade.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ const TRADE_TIMEOUT_SECONDS = 300;
 const { concat, hexlify, hexValue } = ethers.utils;
 
 export async function makeTrade(
-  tokenListUrl: string,
+  tokenListUrl: string | null,
```
Comment (nlordell) [inline src/make_trade.ts:34]: Is this really null or undefined?
```suggestion
  tokenListUrl: string | undefined,
```

Comment (nlordell) [reply src/make_trade.ts:34]: **edit**: Tried it out, and indeed it is `undefined` when un-set and not `null`:

```ts
task("foobar", "Foo Bar")
  .addOptionalParam("param", "Some param")
  .setAction(async ({ param }) => console.log(param));
```

Will print out `undefined` when no parameter is specified.

File: src/utils.ts [hunk 3]
```diff
@@ -13,7 +13,18 @@ export class ChainUtils {
     } else if (network.config.chainId === 100) {
       return Chain.XDAI;
     } else {
-      throw `Unsexpected network ${network.config.chainId}`;
+      throw `Unexpected network ${network.config.chainId}`;
```
Comment (nlordell) [inline src/utils.ts:16]: nit: (sorry missed this before)
```suggestion
      throw new Error(`Unexpected network ${network.config.chainId}`);
```

In general throwing errors is preferable to `string` as it provides properties like `message` (the specified error message) and more importantly `stacktrace`, which is a captured stacktrace from where the error is thrown and makes debugging easier.

Consider the code:
```js
function foo() { bar(); }
function bar() { baz(); }
function baz() { throw ... }
foo();
```

Throwing a `string`:
```
$ node ./foo_string.js

/var/home/nlordell/Developer/gp-v2-contracts/foo_string.js:3
function baz() { throw "error" }
                 ^
error
(Use `node --trace-uncaught ...` to show where the exception was thrown)
```

Throwing an `Error`:
```
$ node ./foo_error.js   
/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:3
function baz() { throw new Error("error") }
                 ^

Error: error
    at baz (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:3:24)
    at bar (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:2:18)
    at foo (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:1:18)
    at Object.<anonymous> (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:4:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

Comment (nlordell) [reply src/utils.ts:16]: If you don't want to show the stack trace, you can always catch errors at a top level and log only the message, and then print the stack trace to some debug logs (if using the `debug` package for example).

This also has the advantage that if there is some uncaught exception in some fiber that isn't `await`-ed you will get a stack trace.

**Review by nlordell — APPROVED** [review]
Looks good, just some small nits.

**Pull request merged**